src/net/mod.rs

```diff
@@ -62,6 +78,44 @@ mod conn {
     }
 }
 
+#[cfg(target_arch = "wasm32")]
+mod wasm_connector {
+    use super::*;
+    use async_trait::async_trait;
+    use std::io::Error as IoError;
+    use ws_stream_wasm::WsMeta;
+    #[derive(Clone, Default)]
+    pub struct DefaultDomainWebsocketConnector {}
+    impl DefaultDomainWebsocketConnector {
+        pub fn new() -> Self {
+            Self {}
+        }
+    }
+    #[async_trait(?Send)]
+    impl TcpDomainConnector for DefaultDomainWebsocketConnector {
+        async fn connect(
+            &self,
+            addr: &str,
+        ) -> Result<(BoxWriteConnection, BoxReadConnection, ConnectionFd), IoError> {
```

Comment:
I think we might need to move to a generic error type for this rather than `IoError` to allow for describing the websocket errors.